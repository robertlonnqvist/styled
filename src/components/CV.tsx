import React from "react";
import Container from "./Container";

const keywords = [
  "Java",
  "Javascript/ECMAScript",
  "Typescript",
  "Ruby",
  "Python",
  "Go",

  "NodeJS",
  "HTML",
  "CSS",
  "NPM",
  "jQuery",
  "React",
  "Angular",
  "Webpack",
  "Parcel",

  "Rest",
  "GraphQL",
  "Protobuf",
  "Soap",

  "Spring Boot",
  "Spring Core",
  "Spring MVC",
  "Maven",
  "Gradle",
  "Flyway",
  "JavaEE",
  "Jersey",
  "JAXWS",
  "JSP",
  "JSF",
  "Thymeleaf",
  "SQL",
  "Tomcat",
  "Glassfish",
  "JBOSS/Wildfly",

  "Shell",
  "Bash",
  "Git",
  "SVN",

  "MacOS",
  "Linux",
  "Windows",

  "MySQL",
  "MariaDB",
  "PostgreSQL",
  "SQL Server",
  "Redis",
  "MongoDB",

  "Visual Studio Code",
  "IntelliJ IDEA",
  "VIM",
  "Eclipse",
  "NetBeans",

  "AWS",
  "Azure",
  "Kibana",
  "APM",
  "SonarQube",
  "Kafka",
  "Gitlab",
  "GitlabCI",
  "GitHub",
  "GitHub Actions",
  "Jenkins",
  "ContentFul",
  "Netifly",
  "Docker",
];

const experiences = [
  {
    title: "2020- Tele2 Sverige AB, System Architect",
    body: `System architect and tech lead for six development teams. My role is to
        support the teams with techical decisions and bridge between the
        enterprise architects and the developers making sure we are doing the
        right things.`,
  },
  {
    title: "2019-2020 Svenska Spel AB, Software Developer",
    body: `Software developer part of a small development team refactoring a big
        Java based system. The system was deployed on ~2000 devices accross
        Sweden. The work included modernize the code base and refactor legacy
        code introducing IoC with Spring framework.`,
  },
  {
    title: "2018-2019 Tele2 Sverige AB, Domain Architect",
    body: `Domain architect and responsible for web, self-service and customer
        integrations at TELE2 B2B IT Services. Working with TELE2 online
        channels for self-service for B2B (SME and LE) segments in customer
        projects, IT infrastructure and product management. Project screening
        and solution design.`,
  },
  {
    title: "2017-2018 Tele2 Sverige AB, Solution Architect",
    body: `Architect for online channels and integration services at Tele2 B2B IT
        Services. Working with solutions to migrate and integrate TDC
        application stack onto Tele2 application IT infrastructure. Worked in
        agile projects and end customer projects as well as product management.
        Responsibilities in both design and implementation.`,
  },
  {
    title: "2014-2017 TDC Sverige AB, IT Architect",
    body: `Architect for online channels and integration services in the Nordics.
        Responsible for TDC Nordics web channels and OSS (Operational Service
        Software) applications. Worked in agile projects with product owners and
        product management as well as end customers. Responsibilities in both
        design and implementation.`,
  },
  {
    title: "2007-2014 TDC Sverige AB, System Architect",
    body: `Development lead for TDC's self-service customer portal Nordic TDC
        Service Online. Responsible for a team of 5-10 developers working
        closely towards product managers and product owner. Participated in
        implementation and transition of SCRUM for the development teams.`,
  },
  {
    title: "2006-2007 TDC Song AB, Software Developer",
    body: `Developer for TDC's self-service customer portal Nordic TDC Service
        Online. Member of a development team consisting of 5-10 developers. The
        application consisted of a PHP based application which was migrated to
        Java with Spring MVC.`,
  },
];

const CV = (): JSX.Element => {
  return (
    <Container title="CV" headline="Curriculum vitae">
      <h2 className="text-xl text-blue-400 font-bold mb-3">Technical skills</h2>
      <p className="mb-3">
        <i> {keywords.join(", ")}... and probably more I forgot about :)</i>
      </p>
      <h2 className="text-xl text-blue-400 font-bold mb-3">Experience</h2>
      {experiences.map(({ title, body }, i) => (
        <div key={i}>
          <h3 className="text-lg text-yellow-400 font-bold mb-3">{title}</h3>
          <p className="mb-3">{body}</p>
        </div>
      ))}
    </Container>
  );
};

export default CV;
