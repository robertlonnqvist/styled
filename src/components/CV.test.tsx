import React from "react";
import { render } from "@testing-library/react";
import CV from "./CV";
import { MemoryRouter } from "react-router-dom";

test("renders CV component", () => {
  const { container } = render(
    <MemoryRouter>
      <CV />
    </MemoryRouter>
  );
  expect(container).toMatchSnapshot();
});
