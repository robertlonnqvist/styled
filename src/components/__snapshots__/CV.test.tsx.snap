// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CV component 1`] = `
<div>
  <div
    class="container mx-auto max-w-4xl p-1 md:p-10"
  >
    <header
      class="flex justify-between flex-col md:flex-row items-center"
    >
      <a
        class="font-extrabold text-2xl hover:text-white mb-2"
        href="/"
      >
        <span
          class="text-pink-600 "
        >
          &gt;&gt;
        </span>
         Robert LÃ¶nnqvist
      </a>
      <nav
        class="flex justify-around space-x-2 items-center mb-2"
      >
        <a
          class="font-medium block px-4 py-2 hover:text-white hover:bg-gray-500 rounded-lg "
          href="/"
        >
          About
        </a>
        <a
          class="font-medium block px-4 py-2 hover:text-white hover:bg-gray-500 rounded-lg bg-gray-500 text-white"
          href="/cv"
        >
          CV
        </a>
        <a
          class="font-medium block px-4 py-2 hover:text-white hover:bg-gray-500 rounded-lg"
          href="https://github.com/robertlonnqvist"
        >
          Code
        </a>
        <a
          class="font-medium block px-4 py-2 hover:text-white hover:bg-gray-500 rounded-lg"
          href="https://www.linkedin.com/in/robertlonnqvist"
        >
          Contact
        </a>
      </nav>
    </header>
    <div
      class="flex flex-col bg-gray-800 rounded-lg shadow-lg"
    >
      <div
        class="flex justify-between items-center px-4 py-2 bg-black rounded-t-lg"
      >
        <div
          class="flex justify-start space-x-2"
        >
          <div
            class="w-3 h-3 bg-red-500 rounded-full"
          />
          <div
            class="w-3 h-3 bg-yellow-400 rounded-full"
          />
          <div
            class="w-3 h-3 bg-green-400 rounded-full"
          />
        </div>
        <div
          class="text-center flex-grow"
        >
          CV
        </div>
      </div>
      <div
        class="m-3"
      >
        <h1
          class="text-2xl text-green-400 mb-3"
        >
          Curriculum vitae
        </h1>
        <h2
          class="text-xl text-blue-400 mb-3"
        >
          Technical skills
        </h2>
        <p
          class="mb-3"
        >
          <i>
             
            Java, Javascript/ECMAScript, Typescript, Ruby, Python, Go, NodeJS, HTML, CSS, NPM, jQuery, React, Angular, Webpack, Parcel, Rest, GraphQL, Protobuf, Soap, Spring Boot, Spring Core, Spring MVC, Maven, Gradle, Flyway, JavaEE, Jersey, JAXWS, JSP, JSF, Thymeleaf, SQL, Tomcat, Glassfish, JBOSS/Wildfly, Shell, Bash, Git, SVN, MacOS, Linux, Windows, MySQL, MariaDB, PostgreSQL, SQL Server, Redis, MongoDB, Visual Studio Code, IntelliJ IDEA, VIM, Eclipse, NetBeans, AWS, Azure, Kibana, APM, SonarQube, Kafka, Gitlab, GitlabCI, GitHub, GitHub Actions, Jenkins, ContentFul, Netifly, Docker
            ... and probably more I forgot about :)
          </i>
        </p>
        <h2
          class="text-xl text-blue-400 mb-3"
        >
          Experience
        </h2>
        <div>
          <h3
            class="text-lg text-yellow-400 mb-3"
          >
            2020- Tele2 Sverige AB, System Architect
          </h3>
          <p
            class="mb-3"
          >
            System architect and tech lead for six development teams. My role is to
        support the teams with techical decisions and bridge between the
        enterprise architects and the developers making sure we are doing the
        right things.
          </p>
        </div>
        <div>
          <h3
            class="text-lg text-yellow-400 mb-3"
          >
            2019-2020 Svenska Spel AB, Software Developer
          </h3>
          <p
            class="mb-3"
          >
            Software developer part of a small development team refactoring a big
        Java based system. The system was deployed on ~2000 devices accross
        Sweden. The work included modernize the code base and refactor legacy
        code introducing IoC with Spring framework.
          </p>
        </div>
        <div>
          <h3
            class="text-lg text-yellow-400 mb-3"
          >
            2018-2019 Tele2 Sverige AB, Domain Architect
          </h3>
          <p
            class="mb-3"
          >
            Domain architect and responsible for web, self-service and customer
        integrations at TELE2 B2B IT Services. Working with TELE2 online
        channels for self-service for B2B (SME and LE) segments in customer
        projects, IT infrastructure and product management. Project screening
        and solution design.
          </p>
        </div>
        <div>
          <h3
            class="text-lg text-yellow-400 mb-3"
          >
            2017-2018 Tele2 Sverige AB, Solution Architect
          </h3>
          <p
            class="mb-3"
          >
            Architect for online channels and integration services at Tele2 B2B IT
        Services. Working with solutions to migrate and integrate TDC
        application stack onto Tele2 application IT infrastructure. Worked in
        agile projects and end customer projects as well as product management.
        Responsibilities in both design and implementation.
          </p>
        </div>
        <div>
          <h3
            class="text-lg text-yellow-400 mb-3"
          >
            2014-2017 TDC Sverige AB, IT Architect
          </h3>
          <p
            class="mb-3"
          >
            Architect for online channels and integration services in the Nordics.
        Responsible for TDC Nordics web channels and OSS (Operational Service
        Software) applications. Worked in agile projects with product owners and
        product management as well as end customers. Responsibilities in both
        design and implementation.
          </p>
        </div>
        <div>
          <h3
            class="text-lg text-yellow-400 mb-3"
          >
            2007-2014 TDC Sverige AB, System Architect
          </h3>
          <p
            class="mb-3"
          >
            Development lead for TDC's self-service customer portal Nordic TDC
        Service Online. Responsible for a team of 5-10 developers working
        closely towards product managers and product owner. Participated in
        implementation and transition of SCRUM for the development teams.
          </p>
        </div>
        <div>
          <h3
            class="text-lg text-yellow-400 mb-3"
          >
            2006-2007 TDC Song AB, Software Developer
          </h3>
          <p
            class="mb-3"
          >
            Developer for TDC's self-service customer portal Nordic TDC Service
        Online. Member of a development team consisting of 5-10 developers. The
        application consisted of a PHP based application which was migrated to
        Java with Spring MVC.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
